packages/sewing-kit-koa/src/assets.ts

```diff
@@ -52,24 +31,22 @@ enum AssetKind {
   Scripts = 'js',
 }
 
-const DEFAULT_MANIFEST_PATH = 'build/client/assets.json';
-
 export default class Assets {
   assetPrefix: string;
   userAgent?: string;
-  manifestPath: string;
+  private manifests: Manifests;
   private resolvedManifestEntry?: Manifest;
 
   constructor({assetPrefix, userAgent, manifestPath}: Options) {
```

Comment:
Given that we create[ one instance of this per request](https://github.com/Shopify/quilt/blob/master/packages/sewing-kit-koa/src/middleware.ts#L36) is there a reason we don't add `locale` to the class itself? Then we could remove the params all over the place for `locale` and just use the class attribute. Feels weird that we have a mix of params and class attributes for things that are per request.

Reply:
Yah, this is nasty ☹️  It's an artefact of [how userAgent is available when the middleware creates Assets](https://github.com/Shopify/quilt/blob/5ba6cd69793a071950467c5f09c4dee9e93d15d0/packages/sewing-kit-koa/src/middleware.ts#L35-L45), but our apps tend to fetch locale info via GraphQL.

Locales are a niche feature right now, and I figure it's better to just add this verbose version of things before we put any more work into the API.